src/clj/puppetlabs/services/jruby/jruby_puppet_schemas.clj

```diff
@@ -108,7 +115,8 @@
    :borrow-timeout schema/Int
    :max-active-instances schema/Int
    :max-requests-per-instance schema/Int
-   :use-legacy-auth-conf schema/Bool})
+   :use-legacy-auth-conf schema/Bool
+   (schema/optional-key :environment-class-cache-enabled) schema/Bool})
```

Comment:
I think it needs to be in the schema, otherwise it will break the jruby-puppet service (because it will still be in that section). I guess we could dissoc it from the config in https://github.com/rlinehan/puppet-server/blob/SERVER-1185-move-setting-to-jruby-puppet/src/clj/puppetlabs/services/jruby/jruby_puppet_core.clj#L174 first?